lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -2138,24 +2138,24 @@ namespace Js
         public:
             AutoRestoreFunctionInfo(ParseableFunctionInfo *pfi) : pfi(pfi), funcBody(nullptr) {}
             ~AutoRestoreFunctionInfo() {
-                if (this->funcBody && this->funcBody->GetFunctionInfo()->GetFunctionProxy() == this->funcBody)
+                if (this->pfi != nullptr)
                 {
-                    FunctionInfo *functionInfo = funcBody->GetFunctionInfo();
+                    FunctionInfo *functionInfo = this->pfi->functionInfo;
```

Comment:
Before we (re-)set these attributes, can we confirm that this->funcBody is non-null and that functionInfo->GetFunctionProxy()==this->funcBody? (We can still get functionInfo from this->pfi in all cases.)